vampire.cpp

```diff
@@ -168,9 +172,9 @@ Problem *doProving()
  * Read a problem and output profiling information about it.
  * @since 03/08/2008 Torrevieja
  */
-void profileMode()
+void profileMode(Problem* problem)
 {
-  ScopedPtr<Problem> prb(UIHelper::getInputProblem(*env.options));
+  ScopedPtr<Problem> prb(problem);
```

Comment:
If we are `ScopedPtr`ing things, maybe most methods should take a `const Problem &` and create new `ScopedPtr<Problem>`s if they need to?

Understand if this is "future work".

Reply:
`ScopedPointer` is like `unique_ptr`, right? So this is a feeble attempt at saying "we care about reclaiming the memory" when done. 

In that light
> and create new ScopedPtr<Problem>s if they need to?
sounds a bit scary. Copying a problem (does it own it's units, no it doesn't now, but it did until recently) could be tricky...

Reply:
Ugh, I see what you mean. Long-term we should out all these ownership things. But for this let's not.